digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/HistoryDAO.scala

```diff
@@ -0,0 +1,198 @@
+package fi.liikennevirasto.digiroad2.dao
+
+import slick.driver.JdbcDriver.backend.Database
+import Database.dynamicSession
+import fi.liikennevirasto.digiroad2.asset.{AssetTypeInfo, Manoeuvres, ServicePoints}
+import slick.jdbc.StaticQuery.interpolation
+
+object HistoryDAO {
+  val standardTableValues = Seq("DATE_PERIOD_VALUE","ADDITIONAL_PANEL", "VALIDITY_PERIOD_PROPERTY_VALUE",
+    "DATE_PROPERTY_VALUE", "SERVICE_POINT_VALUE", "NUMBER_PROPERTY_VALUE", "MULTIPLE_CHOICE_VALUE",
+    "SINGLE_CHOICE_VALUE", "TEXT_PROPERTY_VALUE", "PROHIBITION_VALUE")
+
+  def getExpiredAssetsIdsByAssetTypeAndYearGap(assetType: AssetTypeInfo, yearGap: Int) = {
+    val yearGapCondition = s"EXTRACT(YEAR FROM VALID_TO) < EXTRACT(YEAR FROM SYSDATE) - $yearGap"
+
+    assetType match {
+      case Manoeuvres =>
+        sql"""
+          SELECT ID FROM MANOEUVRE WHERE #$yearGapCondition
+        """.as[Long].list
+      case _ =>
+        sql"""
+          SELECT ID FROM ASSET WHERE ASSET_TYPE_ID = ${assetType.typeId} AND #$yearGapCondition
+        """.as[Long].list
+    }
+  }
```

Comment:
Can be moved to HistoryService and created two queries to get the values at the DAO

Reply:
Done